src/Raven.Server/Documents/Handlers/TransactionsRecordingHandler.cs

```diff
@@ -0,0 +1,173 @@
+                    id: operationId,
+                    token: operationCancelToken
+                );
+
+                using (ContextPool.AllocateOperationContext(out DocumentsOperationContext context))
+                using (var writer = new BlittableJsonTextWriter(context, ResponseBodyStream()))
+                {
+                    context.Write(writer, result.ToJson());
+                }
+            }
+        }
+
+        private async Task<IOperationResult> DoReplay(
+            Action<IOperationProgress> onProgress,
+            Stream replayStream,
+            CancellationToken token)
+        {
+            const int commandAmountBetweenRespond = 15;
+
+            try
+            {
+                return await Task.Run(() =>
+                {
+                    long commandsProgress = 0;
+                    var stopwatch = Stopwatch.StartNew();
+                    stopwatch.Start();
+                    foreach (var replayProgress in Database.TxMerger.Replay(replayStream))
+                    {
+                        commandsProgress = replayProgress.CommandsProgress;
+                        //TODO Maybe should be relative to time or size
+                        if (replayProgress.CommandsProgress % commandAmountBetweenRespond == 0)
+                        {
+                            onProgress(new ReplayTrxProgress
+                            {
+                                ProcessedCommand = replayProgress.CommandsProgress,
+                                PassedTime = stopwatch.Elapsed
+                            });
+                        }
+
+                        token.ThrowIfCancellationRequested();
+                    }
+                    stopwatch.Stop();
+
+                    return new ReplayTrxOperationResult
+                    {
+                        ExecutedCommandsAmount = commandsProgress,
+                        PassedTime = stopwatch.Elapsed
+                    };
+                }, token);
+            }
+            catch (Exception e)
+            {
+                HttpContext.Response.Headers["Connection"] = "close";
+                throw new InvalidOperationException("Failed to process replay transaction commands", e);
+            }
+
+        }
+
+        [RavenAction("/databases/*/admin/transactions/start-recording", "POST", AuthorizationStatus.ClusterAdmin)]
+        public async Task StartRecording()
+        {
+            using (ContextPool.AllocateOperationContext(out DocumentsOperationContext context))
+            {
+                var json = await context.ReadForMemoryAsync(RequestBodyStream(), null);
+                var parameters = JsonDeserializationServer.StartTransactionsRecordingOperationParameters(json);
+                if (parameters.File == null)
+                {
+                    ThrowRequiredPropertyNameInRequest(nameof(parameters.File));
+                }
+
+                var command = new TransactionsRecordingCommand(
```

Comment:
isn't this an overkill? can we do `Database.TxMerger.Start(parameters.File)` ?

Reply:
We have to do it this way to avoid starting recording in the middle of a transaction.
This way, we always start from a brand new tx.

Reply:
Makes sense.